modules/appeals_api/app/models/appeals_api/notice_of_disagreement.rb

```diff
@@ -152,5 +142,10 @@ def birth_date(who)
     def header_field_as_string(key)
       auth_headers&.dig(key).to_s.strip
     end
+
+    # After expunging pii, form_data is nil, update will fail unless validation skipped
```

Comment:
I can remove this, but I really feel like leaving this is the nice thing to do for anyone in the future coming to the code.

Reply:
Have you verified this works as expected? My understanding with the `validate(...if:...)` was that it checks the type passed into the `if` and knows to call `self` as the passed object into the proc. By changing it to a symbol & method call, I'm not confident the `proc` inside `pii_present?` will do what you expect - it has no context for what `a` is, in this case.

I could totally be wrong about that, of course, it just looks odd to me (e.g. instead of doing a `self.form_data.present?`) and I want to make sure :)

Reply:
I'll go back and check the context inside the new method. I do really like naming it what it actually is. I found that proc to be so confusing for so long until I finally dug into the removal of pii. As mentioned before, I just think it needs clarity, so I have no problem putting the proc back with the validation, but I'm gonna keep pushing for a comment in the code.

Reply:
It works :) Variable `a` is an instance of NOD and accurately returns true or false depending on the presence of form_data.

Reply:
I don't really see a reason to use a `proc` here, though. Why don't we just call `form_data` on self?

`form_data.present?`